src/Symfony/Component/Mailer/Bridge/Amazon/Credential/InstanceCredentialProvider.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Bridge\Amazon\Credential;
+
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\Mailer\Exception\RuntimeException;
+
+/**
+ * Based on: aws-sdk-php / Credentials/InstanceProfileProvider.php.
+ *
+ * @author Karoly Gossler <connor@connor.hu>
+ */
+class InstanceCredentialProvider
+{
+    const SERVER_URI = 'http://169.254.169.254';
+    const CREDENTIAL_PATH = '/latest/meta-data/iam/security-credentials/';
+
+    public function __construct(int $retries = 3)
+    {
+        $this->retries = $retries;
+        $this->client = HttpClient::create();
```

Comment:
Those 2 variables should be declared in the class.

Reply:
For the `HttpClient`, we should allow to inject one and pass the one provided to the `SesTransport`. If none is passed, we should check that `HttpClient` is available:

```php
        if (null === $client) {
            if (!class_exists(HttpClient::class)) {
                throw new LogicException(sprintf('You cannot use "%s" as the HttpClient component is not installed. Try running "composer require symfony/http-client".', __CLASS__));
            }

            $this->client = HttpClient::create();
        }

```